instrumentation/apache-httpclient/apache-httpclient-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/async/ApacheHttpAsyncClientHttpAttributesGetter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachehttpclient.async;
+
+import static io.opentelemetry.javaagent.instrumentation.apachehttpclient.async.ApacheHttpClientRequest.headersToList;
+import static java.util.logging.Level.FINE;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesGetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.ProtocolVersion;
+
+final class ApacheHttpAsyncClientHttpAttributesGetter
+    implements HttpClientAttributesGetter<ApacheHttpClientRequest, HttpResponse> {
+  private static final Logger logger =
+      Logger.getLogger(ApacheHttpAsyncClientHttpAttributesGetter.class.getName());
+
+  @Override
+  public String method(ApacheHttpClientRequest request) {
+    return request.getMethod();
+  }
+
+  @Override
+  public String url(ApacheHttpClientRequest request) {
+    return request.getUrl();
+  }
+
+  @Override
+  public List<String> requestHeader(ApacheHttpClientRequest request, String name) {
+    return request.getHeader(name);
+  }
+
+  @Override
+  public Long requestContentLength(
+      ApacheHttpClientRequest request, @Nullable HttpResponse response) {
+    return request.requestContentLength();
+  }
+
+  @Override
+  @Nullable
+  public Long requestContentLengthUncompressed(
+      ApacheHttpClientRequest request, @Nullable HttpResponse response) {
+    return null;
+  }
+
+  @Override
+  public Integer statusCode(ApacheHttpClientRequest request, HttpResponse response) {
+    return response.getCode();
+  }
+
+  @Override
+  @Nullable
+  public String flavor(ApacheHttpClientRequest request, @Nullable HttpResponse response) {
```

Comment:
if this is exactly the same as method in classic instrumentation, how about extracting common helper somewhere?

Reply:
More or less it is same but the request type is different, I guess we will have to keep 2 classes.